examples/planewave-quad-comparison.py

```diff
@@ -0,0 +1,197 @@
+"""
+Tensor Product Quadrature vs. Vioreanu-Rokhlin Quadrature for Plane Wave on Sphere
+==================================================================================
+
+This test compares the absolute error of **Tensor Product Quadrature** and
+**Vioreanu-Rokhlin Quadrature** against the number of discretization nodes
+with matched total polynomial degree exactness.
+
+Comparison of Polynomial Exactness
+----------------------------------
+
+The following table summarizes the total degree of polynomial exactness of both
+quadrature methods based on the order:
+
+Order     VR Exact_to     Tensor Exact_to
+-----------------------------------------
+1         2              3
+2         4              5
+3         5              7
+4         7              9
+5         8              11
+6         10             13
+7         12             15
+8         14             17
+9         15             19
+10        17             21
+11        19             23
+12        20             25
+13        22             27
+14        24             29
+15        25             31
+16        27             33
+17        28             35
+18        30             37
+19        32             39
+
+
+Wave Function and Sphere Integral
+---------------------------------
+
+The normal direction is:
+
+    d = [-5, 4, 1], n = d / <d, d>
+
+The plane wave is defined as:
+
+    f(x) = exp(1j * n · x)
+
+We compute the integral of the plane wave over a sphere of radius 1:
+
+    ∫_sphere f dS ≈ 10.57423625632583807548
+
+This value is obtained using Mathematica with a working precision of 21 digits.
+
+Mathematica Code
+----------------
+
+Below is the Mathematica code used to define the wave function and
+compute the integral numerically:
+
+    n = Normalize[{-5, 4, 1}];
+    r = 1;
+    wave[θ_, φ_] :=
+      Exp[I r (n[[1]] Sin[θ] Cos[φ] +
+               n[[2]] Sin[θ] Sin[φ] +
+               n[[3]] Cos[θ])];
+
+    NIntegrate[
+     wave[θ, φ] * r^2 Sin[θ],
+     {θ, 0, Pi},
+     {φ, 0, 2 Pi},
+     WorkingPrecision -> 21
+    ]
+
+"""
+
+import meshmode.mesh.generation as mgen
+import numpy as np
+import matplotlib.pyplot as plt
+from meshmode.discretization import Discretization
+from meshmode.discretization.poly_element import InterpolatoryQuadratureGroupFactory
+from meshmode.mesh import (
+    SimplexElementGroup,
+    TensorProductElementGroup,
+)
+from pytential.qbx import QBXLayerPotentialSource
+from pytential import GeometryCollection, bind, sym
+from arraycontext import flatten
+from meshmode.array_context import PyOpenCLArrayContext
+import pyopencl as cl
+
+cl_ctx = cl.create_some_context()
+queue = cl.CommandQueue(cl_ctx)
+actx = PyOpenCLArrayContext(queue)
+
+
+def quadrature(level, target_order, qbx_order, group_cls=SimplexElementGroup):
```

Comment:
The way this is currently expressed is flawed: At no point do you explicitly specify which quadrature should be used. And what pytential uses internally is an implementation detail that could change at any time. If you're intending to measure differences between specific quadrature rules, your code should exert positive control over what rule is used, instead of relying on implementation coincidence. It may be easier to do this by using meshmode primitives.